pairing/src/lib.rs

```diff
@@ -141,20 +175,30 @@ impl PyG1 {
         let g =  G1::one();
         obj.init(|t| PyG1{
             g1: g,
+            pp: Vec::new(),
+            pplevel : 0
         })
     }
 
     fn rand(&mut self, s1: u32, s2: u32, s3: u32, s4: u32) -> PyResult<()>{
         let mut rng = XorShiftRng::from_seed([s1,s2,s3,s4]);
         let g = G1::rand(&mut rng);
         self.g1 = g;
+        if self.pplevel != 0{
```

Comment:
I think it would be cleaner if you added a space after the conditional check in the `if` statements--
```suggestion
if self.pplevel != 0 {
...
}
```